Apply fixes to compile under MacOSX

palmerj opened the pull request.

palmerj committed:
Apply fixes to compile under MacOSX

palmerj committed:
Apply fixes to compile under MacOSX

palmerj commented:
Low priority. Just to allow me to read BDE file on my mac

palmerj commented:
Looks like the travis errors are not related to this PR - dpkg-buildpackage -b -us -uc is returning an error.

strk commented on src/bde_copy.cpp:
```diff
@@ -401,7 +402,7 @@ void set_fields( const char *fields )
         char *type = name + strlen(name);
         char *p = strchr(name,'=');
-        if( p > 0 )
+        if( p > (void *)0 )
```
This should be equivalent to NULL

palmerj replied:
Fixed.

strk commented:
@imincik the failure seems related to Debian package building:
```
dpkg-checkbuilddeps: Unmet build dependencies: cmake
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
```

Something must have changed on the travis side because things worked on previous build...

strk committed:
Explicitly install cmake

Mostly a test for
https://github.com/linz/linz-bde-copy/pull/35#issuecomment-344127487

strk commented:
`cmake` is listed in `Build-Depends` of debian/control so I've no idea where is that error coming from. Let's see how #36 works...

strk commented:
#36 passes, can you see if merging that one in your PR fixes this too ? Or you'd have to rebase after that one is merged anyway...

palmerj committed:
Use non-equals operator

palmerj committed:
Resolved merge conflict

palmerj commented:
I've rolled in #36 to this PR. Lets see if the build is ok

palmerj commented:
All tests passed. If you are happy please merge

strk commented:
happily merged :)

palmerj commented:
Thanks :)
